services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -1754,8 +1819,24 @@ class ApplicationServiceITest : DatabaseTest() {
     private fun mockApplicationWithArea(
         applicationData: ApplicationData =
             createCableReportApplicationData(areas = listOf(aleksanterinpatsas)),
-        alluId: Int? = null
-    ): Application = createApplication(alluid = alluId, applicationData = applicationData)
+        alluId: Int? = null,
+        applicationIdentifier: String? = null,
+    ): Application =
+        createApplication(
+            alluid = alluId,
+            applicationIdentifier = applicationIdentifier,
+            applicationData = applicationData
+        )
+
+    private fun mockSendAllu(applicationEntity: ApplicationEntity) {
```

Comment:
This doesn't mock anything, at least not in the way we usually mean when talking about mocking in tests.

Reply:
Renamed to fakeSendAllu, could not figure out anything better.

Reply:
How about setAlluFields?

Even better would be if the application was created with these values set, so there's no need to update them afterwards.

Reply:
Maybe something like:
```
            val initialApplicationData =
                createCableReportApplicationData(
                    customerWithContacts = hakijaCustomerContact,
                    contractorWithContacts = suorittajaCustomerContact,
                )
            val application =
                alluDataFactory.saveApplicationEntity(
                    USERNAME,
                    hanke = createHankeEntity(hankeTunnus = HANKE_TUNNUS),
                    application = mockApplicationWithArea()
                ) {
                    it.alluid = 21
                    it.alluStatus = PENDING
                    it.applicationIdentifier = defaultApplicationIdentifier
                    it.applicationData = initialApplicationData
                }
```

Reply:
This will not create the initial hankekayttaja (perustaja) which is required for sending emails. Using generateHankeWithApplication is not ideal but it does the job.

Renamed the problematic function to setAlluFields.